samza-azure/src/main/java/org/apache/samza/coordinator/AzureCoordinationUtils.java

```diff
@@ -52,6 +52,16 @@ public DistributedLockWithState getLockWithState(String lockId) {
     return new AzureLock(blob);
   }
 
+  @Override
+  public DistributedReadWriteLock getReadWriteLock(String lockId) throws UnsupportedOperationException {
```

Comment:
getReadWriteLock() is used in LocalApplicationRunner (LAR) when app.mode=BATCH. If we throw an exception here instead of returning null, none of the standalone deployments for BATCH mode using Azure will even work. 

Currently, no run.id is generated for standalone (for both ZK and Azure) and thus, if we return null here instead of an exception, it would revert to the existing behavior until we support run.id for Azure.

Would it help if we log a warning explaining that run.id will not be generated (meaning previous state will be reused) and this batch job wont starting with fresh state?